db.go

```diff
@@ -0,0 +1,97 @@
+package money
+
+import (
+	"database/sql/driver"
+	"fmt"
+	"strconv"
+	"strings"
+)
+
+// Scan implements sql.Scanner to deserialize a Money instance from an "amount,currency_code" comma-separated pair
+func (m *Money) Scan(src interface{}) error {
+	amount:= &Amount{}
+	currency:= &Currency{}
+
+	// let's support string and int64
+	switch src.(type) {
+	case string:
+		parts := strings.Split(src.(string), ",")
+		if len(parts) != 2 || parts[0] == "" || parts[1] == "" {
+			return fmt.Errorf("%#v is not valid to scan into Money; update your query to return a comma-separated pair of \"amount,currency_code\"", src.(string))
+		}
+		if err := amount.Scan(parts[0]); err != nil {
+			return fmt.Errorf("scanning %#v into an Amount: %v", parts[0], err)
+		}
+		if err := currency.Scan(parts[1]); err != nil {
+			return fmt.Errorf("scanning %#v into a Currency: %v", parts[1], err)
+		}
+	default:
+		return fmt.Errorf("don't know how to scan %T into Money; update your query to return a comma-separated pair of \"amount,currency_code\"", src)
+	}
+
+	// allocate new Money with the scanned amount and currency
+	*m = Money{
+		amount: amount,
+		currency: currency,
+	}
+
+	return nil
+}
+
+// Value implements driver.Valuer to serialize an Amount into sql/driver compatible type
+func (a Amount) Value() (driver.Value, error) {
+	return a.val, nil
```

Comment:
"val" was removed from Amount

Reply:
Note: in the last change, Amount was converted to an alias of "int64".
`
type Amount = int64
` 
For this method to work you will need to change it to a new type:
`type Amount int64
`
(Removing the "=" sign)

Reply:
this will simplify the `db.go` file as the `driver.Valuer` and `sql.Scanner` interfaces are already implemented for the base `int64` type.

I am planning to rebase my branch on top of the recent changes and handle the Amount updates that way.